src/machine/machine_avr.go

```diff
@@ -157,3 +157,96 @@ func (a ADC) Get() uint16 {
 	high := uint16(*avr.ADCH)
 	return uint16(low) | uint16(high<<8)
 }
+
+// I2C on the Arduino
+
+// I2CInit initializes the I2C interface.
+func I2CInit() {
+	// Activate internal pullups for twi.
+	*avr.PORTC |= (avr.DIDR0_ADC4D | avr.DIDR0_ADC5D)
+
+	// Initialize twi prescaler and bit rate.
+	*avr.TWSR |= (avr.TWSR_TWPS0 | avr.TWSR_TWPS1)
+
+	// twi bit rate formula from atmega128 manual pg. 204:
+	// SCL Frequency = CPU Clock Frequency / (16 + (2 * TWBR))
+	// NOTE: TWBR should be 10 or higher for master mode.
+	// It is 72 for a 16mhz board with 100kHz TWI
+	*avr.TWBR = avr.RegValue(72)
+
+	// Enable twi module.
+	*avr.TWCR = avr.TWCR_TWEN
+}
+
+// I2CStart starts a communication session.
+func I2CStart() {
+	// Clear TWI interrupt flag, put start condition on SDA, and enable TWI.
+	*avr.TWCR = (avr.TWCR_TWINT | avr.TWCR_TWSTA | avr.TWCR_TWEN)
+
+	// Wait till start condition is transmitted.
+	for ok := true; ok; ok = (*avr.TWCR & avr.TWCR_TWINT) == 0 {
```

Comment:
Why not just `for (*avr.TWCR & avr.TWCR_TWINT) == 0 {`?
Also, calling it `ok` is a bit misleading: as long as `ok` is true, the condition is not transmitted (it really means 'not ok').

(This also applies to similar statements below).